packages/block-library/src/playlist/edit.js

```diff
@@ -0,0 +1,204 @@
+/**
+ * External Dependencies
+ */
+
+import { uniqBy } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import {
+	FormFileUpload,
+	IconButton,
+	Placeholder,
+	ServerSideRender,
+	Toolbar,
+	withNotices,
+	CheckboxControl,
+	PanelBody,
+	SelectControl,
+} from '@wordpress/components';
+import { Component, Fragment } from '@wordpress/element';
+import {
+	BlockControls,
+	InspectorControls,
+	mediaUpload,
+	MediaUpload,
+} from '@wordpress/editor';
+
+class PlaylistEdit extends Component {
+	constructor() {
+		super( ...arguments );
+
+		this.initializePlaylist = this.initializePlaylist.bind( this );
+		this.uploadFromFiles = this.uploadFromFiles.bind( this );
+		this.onUploadFiles = this.onUploadFiles.bind( this );
+
+		// check for if ids is set to determine edit state
+		this.state = {
+			isEditing: ! this.props.attributes.ids,
+		};
+	}
+
+	initializePlaylist() {
+		window.wp.playlist.initialize();
+	}
+
+	uploadFromFiles( event ) {
+		this.onUploadFiles( event.target.files );
+	}
+
+	onUploadFiles( files ) {
+		function getMimeBaseType( type ) {
+			return type.split( '/' )[ 0 ];
+		}
+
+		const { setAttributes, noticeOperations } = this.props;
+		const isConsistentType = uniqBy( files, ( file ) => getMimeBaseType( file.type ) ).length === 1;
+		if ( ! isConsistentType ) {
+			noticeOperations.createErrorNotice( 'Cannot have mixed types in a Playlist Block' );
+			setAttributes( { ids: undefined, type: undefined } );
+			return;
+		}
+
+		mediaUpload( {
+			allowedType: [ 'audio', 'video' ],
+			filesList: files,
+			onFileChange: ( media ) => {
+				// validate type is consistent for playlist
+				if ( media.length > 0 && media[ 0 ].mimeType && isConsistentType ) {
+					const type = media[ 0 ].mimeType.split( '/' )[ 0 ];
+					const ids = media.map( ( item ) => item.id );
+					setAttributes( { ids, type } );
+					this.setState( { isEditing: false } );
+				}
+			},
+		} );
+	}
+
+	toggleAttribute( attribute ) {
+		return ( newValue ) => {
+			this.props.setAttributes( { [ attribute ]: newValue } );
+		};
+	}
+
+	render() {
+		const { attributes, setAttributes, className, noticeUI } = this.props;
+		const { isEditing } = this.state;
+		const { tracklist, showTrackNumbers, showArtists, images, style, type } = attributes;
+
+		const onSelectMedia = ( media ) => {
+			//check if there are returned media items and set attributes when there are
+			if ( media && media[ 0 ].url ) {
+				const ids = media.map( ( item ) => item.id );
+				setAttributes( { ids, type: media[ 0 ].type } );
+				this.setState( { isEditing: false } );
+			}
+		};
+
+		if ( isEditing ) {
+			return (
+				<Placeholder
```

Comment:
Is it possible to use the `MediaPlaceholder` instead of a custom one like proposed in #9139 This will ensure the media blocks all have the same "design". Or is this very specific?

Reply:
I think this one is specific due to the type validation in the upload form. I could be wrong. I'll see if I can match the MediaPlaceholder arguments. Also, I may need to add some new conditions for `playlist` into MediaPlaceholder... for the sake of getting this merged before we hit conflicts could this be a future issue to fix in a later iteration? I've hit a few merge blockers that cause me a bit of time getting updated. Would love to have a baseline to build off of.

Reply:
I'm fine for this being a follow-up PR :), but I do think we should try to do it for the sake of consistency.